.github/workflows/unit_test.yml

```
@@ -0,0 +1,39 @@
+name: "Unit test"
+# NOTE: We're just skipping the tests requiring earthdata login here; we don't
+#       distinguish yet between unit and integration tests.
+
+on:
+  pull_request:
+  push:
+    branches:
+      - "main"
+      - "development"
+
+
+jobs:
+  test:
+    runs-on: "ubuntu-latest"
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.9", "3.12"]
```

Learning moment (and I know there's an open issue too): why these two versions?

---

I chose the lowest and highest versions of Python that we support. We can test every version we support, but I was trying to balance exhaustiveness with environmental impact. I personally can't think of a problem that would be caught by running our tests for 3.11, but that may just be my ignorance :)

---

Works for me (and makes complete sense).

```suggestion
        python-version: ["3.9", "3.12"] #NOTE: min and max Python versions supported by icepyx
```